src/Parse/Helpers.hs

```diff
@@ -327,7 +327,9 @@ spaces =
 
 forcedWS :: IParser String
 forcedWS =
-  do  ws <- many1 (spaces <|> newline)
+  do  failIfTabFound
+      ws <- many1 (spaces <|> newline)
+      failIfTabFound
```

Comment:
Yes, this is what I tried initially:
```haskell
ws <- many1 (spaces <|> newline <|> (char '\t' >> fail "tab error"))
```

The problem is that it consumes the tab character before failing so the error message position is way out of whack:
```
-- SYNTAX PROBLEM ------------------------------- tests/test-files/good/Tabs.elm

tab error

3│   	[
           ^
Maybe <http://elm-lang.org/docs/syntax> can help you figure it out.
```

To fix that I tried adding `lookAhead`, but that makes it worse. `many1` works in a way that parsers have to fail to indicate that it should stop repeating. If a parser inside `many1` fails *without consuming input* then `many1` returns and let's the next parser take over. 

In short, you cannot put a `lookAhead` inside a `many1` and still propagate an error. This is why `failIfTabFound` has to be outside it.